src/ai/backend/storage/dellemc/__init__.py

```diff
@@ -0,0 +1,139 @@
+from ..vfs import BaseVolume
+from .dellemc_client import DellEMCClient
+from .dellemc_quota_manager import QuotaManager
+
+
+class DellEMCVolume(BaseVolume):
+    endpoint: str
+    dell_admin: str
+    dell_password: str
+
+    async def init(self) -> None:
+        self.endpoint = self.config["dell_endpoint"]
+        self.dell_admin = self.config["dell_admin"]
+        self.dell_password = str(self.config["dell_password"])
+        self.dell_api_version = self.config["dell_api_version"]
+        self.dell_system_name = self.config["dell_system_name"]
+
+        self.dellEMC_client = DellEMCClient(
+            str(self.endpoint),
+            self.dell_admin,
+            self.dell_password,
+            api_version=self.dell_api_version,
+            system_name=self.dell_system_name,
+        )
+
+        self.quota_manager = QuotaManager(
+            str(self.endpoint),
+            self.dell_admin,
+            self.dell_password,
+            api_version=self.dell_api_version,
+        )
+
+    async def shutdown(self) -> None:
+        await self.dellEMC_client.aclose()
+        await self.quota_manager.aclose()
+
+    async def get_capabilities(self) -> FrozenSet[str]:
+        return frozenset([CAP_VFOLDER, CAP_VFHOST_QUOTA, CAP_METRIC])
+
+    async def get_hwinfo(self) -> HardwareMetadata:
+        raw_metadata = await self.dellEMC_client.get_metadata()
+        quotas = await self.quota_manager.list_all_quota()
+        metadata = {"quotas": json.dumps(quotas), **raw_metadata}
+        return {"status": "healthy", "status_info": None, "metadata": {**metadata}}
+
+    async def get_fs_usage(self) -> FSUsage:
+        usage = await self.dellEMC_client.get_usage()
+        return FSUsage(
+            capacity_bytes=usage["capacity_bytes"],
+            used_bytes=usage["used_bytes"],
+        )
+
+    async def get_quota_id(self):
+        quotas = await self.quota_manager.list_all_quota()
+        quota_id = []
+        for quota in quotas:
+            quota_id.append(quota["id"])
+        return quota_id
+
+    async def get_quota(self, vfid: UUID) -> BinarySize:
+        raise NotImplementedError
+
+    async def get_drive_stats(self):
+        resp = await self.dellEMC_client.get_drive_stats()
+
+        if "errors" in resp:
+            raise ExecutionError("api error")
+        return resp
+
+    async def get_protocol_stats(self):
+        resp = await self.dellEMC_client.get_protocol_stats()
+
+        if "errors" in resp:
+            raise ExecutionError("api error")
+        return resp
+
+    async def get_system_stats(self):
+        resp = await self.dellEMC_client.get_system_stats()
+
+        if "errors" in resp:
+            raise ExecutionError("api error")
+        return resp
+
+    async def get_workload_stats(self):
+        resp = await self.dellEMC_client.get_workload_stats()
+
+        if "errors" in resp:
+            raise ExecutionError("api error")
+        return resp
```

Comment:
Could we bubble up more detailed error information if available, in a more structured way?

- Who is the last-resort exception handler? (e.g., middleware in webapps)
- How does it handle any unhandled exceptions? At least we need to leave explicit logs about the full details.